runtime/util/cphelp.c

```diff
@@ -195,21 +195,28 @@ addJarToSystemClassLoaderClassPathEntries(J9JavaVM *vm, const char *filename)
 	UDATA newCount = 0;
 	U_32 entryIndex = 0;
 	J9ClassPathEntry *newEntries = NULL;
-
-	PORT_ACCESS_FROM_JAVAVM(vm);
-
+	J9ClassPathEntry *oldEntries = NULL;
+	UDATA entryCount = 0;
 	UDATA jarPathSize = strlen(filename);
-	J9ClassPathEntry *oldEntries = classLoader->classPathEntries;
-	UDATA entryCount = classLoader->classPathEntryCount;
 	/* PR121889: the classpath strings may be allocated with the J9ClassPathEntry structs. Need to copy them too. */
 	UDATA classPathLength = jarPathSize + 1; /* add space for a terminating null character */
+	J9ClassPathEntry *cpEntry = NULL;
+
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	if (NULL == classLoader->cpEntriesMutex) {
```

Comment:
Can this occur, it was initialized in initializeBootstrapClassPath() ?

Reply:
Yes, on Java 11 and up when there is no bootstrap classpath.